src/Raven.Server/Documents/Indexes/Persistence/Corax/JintCoraxDocumentConverter.cs

```diff
@@ -288,6 +357,12 @@ protected static JsValue TryDetectDynamicFieldCreation(string property, ObjectIn
             }
         }
 
+        if (scope.DynamicFields.TryGetValue(field.Name, out _) == false)
+        {
+            scope.DynamicFields[field.Name] = field.Indexing;
+            scope.CreatedFieldsCount++;
```

Comment:
What do we use it for?

Reply:
When we found new dynamic fields (not found before) we've to persist information on how to process it. Then, after processing the whole doc we can update an indexing mapping for it (and assign the correct analyzer per each).
Take a look here:
https://github.com/ravendb/ravendb/blob/cd0ce04c44454272d85ccf75782a6be6b3bff7fc/src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexWriteOperation.cs#L132-L153

Reply:
ok